cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/KafkaCruiseControlRequestHandler.java

```diff
@@ -0,0 +1,164 @@
+                switch (context.getMethod()) {
+                    case GET_METHOD:
+                        handleGet(context, endPoint, requestConfigOverrides, parameterConfigOverrides);
+                        break;
+                    case POST_METHOD:
+                        handlePost(context, endPoint, requestConfigOverrides, parameterConfigOverrides);
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unsupported request method: " + context.getMethod() + ".");
+                }
+            }
+        } catch (UserRequestException ure) {
+            String errorMessage = handleUserRequestException(ure, context);
+            LOG.error(errorMessage, ure);
+        } catch (ConfigException ce) {
+            String errorMessage = handleConfigException(ce, context);
+            LOG.error(errorMessage, ce);
+        } catch (Exception e) {
+            String errorMessage = handleException(e, context);
+            LOG.error(errorMessage, e);
+        }
+    }
+
+    private void handleGet(CruiseControlRequestContext handler,
+                           CruiseControlEndPoint endPoint,
+                           Map<String, Object> requestConfigOverrides,
+                           Map<String, Object> parameterConfigOverrides)
+            throws Exception {
+        // Sanity check: if the request is for REVIEW_BOARD, two step verification must be enabled.
+        if (endPoint == REVIEW_BOARD && !_cruiseControlEndPoints.twoStepVerification()) {
+            throw new ConfigException(String.format("Attempt to access %s endpoint without enabling '%s' config.",
+                    endPoint, WebServerConfig.TWO_STEP_VERIFICATION_ENABLED_CONFIG));
+        }
+        RequestParameterWrapper requestParameter = requestParameterFor(endPoint);
+        CruiseControlParameters parameters = _cruiseControlEndPoints.config().getConfiguredInstance(requestParameter.parametersClass(),
+                CruiseControlParameters.class,
+                parameterConfigOverrides);
+        if (hasValidParameterNames(handler, parameters)) {
+            requestConfigOverrides.put(requestParameter.parameterObject(), parameters);
+            Request ccRequest = _cruiseControlEndPoints.config().getConfiguredInstance(requestParameter.requestClass(),
+                    Request.class, requestConfigOverrides);
+
+            ccRequest.handle(handler);
+        }
+    }
+
+    private void handlePost(CruiseControlRequestContext handler,
+                            CruiseControlEndPoint endPoint,
+                            Map<String, Object> requestConfigOverrides,
+                            Map<String, Object> parameterConfigOverrides)
+            throws Exception {
+        CruiseControlParameters parameters;
+        RequestParameterWrapper requestParameter = requestParameterFor(endPoint);
+        if (endPoint == REVIEW) {
```

Comment:
Why handlePost check the endPoint with REVIEW, but handleGet check the endpoint with REVIEW_BOARD?

Reply:
We used it like this because, in was used in the same way in the original KafkaCruiseControlServlet.